src/toc.tsx

```diff
@@ -146,164 +175,35 @@ export class TableOfContents extends Widget {
   }
 
   /**
-   * Rerender after showing.
+   * Callback invoked upon an update request.
+   *
+   * @param msg - message
    */
-  protected onAfterShow(msg: Message): void {
+  protected onUpdateRequest(msg: Message): void {
+    // Don't bother if the ToC is not visible...
+    /* if (!this.isVisible) {
```

Comment:
~~Is this commented out for a reason?~~ Edit: I see it was that way already. Feel free to remove, or I can clean it up later if not.

Reply:
Removed.